core/consensus/slots/src/lib.rs

```diff
@@ -230,23 +232,45 @@ pub trait SlotData {
 }
 
 impl SlotData for u64 {
-	fn slot_duration(&self) -> u64 { *self }
+	fn slot_duration(&self) -> u64 {
+		*self
+	}
 
 	const SLOT_KEY: &'static [u8] = b"aura_slot_duration";
 }
 
 /// A slot duration. Create with `get_or_compute`.
 // The internal member should stay private here.
 #[derive(Clone, Copy, Debug, Encode, Decode, Hash, PartialOrd, Ord, PartialEq, Eq)]
-pub struct SlotDuration<T: Clone>(T);
+pub struct SlotDuration<T>(T);
+
+impl<T> Deref for SlotDuration<T> {
+	type Target = T;
+	fn deref(&self) -> &T {
+		&self.0
+	}
+}
+
+impl<T: SlotData + Clone> SlotData for SlotDuration<T> {
+	/// Get the slot duration in milliseconds.
+	fn slot_duration(&self) -> u64
+	where
```

Comment:
```suggestion
		where T: SlotData
```

(follwing line needs to be deleted)